src/mappers.ts

```diff
@@ -1,515 +1,585 @@
+    },
+    secrets: secretsYaml,
+    certificate: certificates,
+    environmentVariables: envYaml,
+    passFailCriteria: {
+      passFailMetrics: failCriteria,
+    },
+    autoAbortCriteria: autoStop,
+    subnetId: subnetId,
+    keyvaultReferenceIdentityType: kvRefType,
+    keyvaultReferenceIdentityId: kvRefId,
+  };
+  return data;
 }
 
 export async function createTestHeader() {
-    let headers: IHeaders = {
-        'content-type': 'application/merge-patch+json',
-        'Authorization': 'Bearer '+ token
-    };
-    return headers;
+  let headers: IHeaders = {
+    "content-type": "application/merge-patch+json",
+    Authorization: "Bearer " + token,
+  };
+  return headers;
 }
 
 export function uploadFileData(filepath: string) {
-    try
-    {
-        let filedata : Buffer = fs.readFileSync(filepath);
-        const readable = new Readable();
-        readable._read = () => {};
-        readable.push(filedata);
-        readable.push(null);
-        return readable;
-    }
-    catch(err:any) {
-        err.message = "File not found "+ filepath;
-        throw new Error(err.message);
-    }
+  try {
+    let filedata: Buffer = fs.readFileSync(filepath);
+    const readable = new Readable();
+    readable._read = () => {};
+    readable.push(filedata);
+    readable.push(null);
+    return readable;
+  } catch (err: any) {
+    err.message = "File not found " + filepath;
+    throw new Error(err.message);
+  }
 }
 
-export async function UploadAndValidateHeader(formData:any) {
-    let headers: IHeaders = {
-        'Authorization': 'Bearer '+ token , 
-        'content-type':'application/octet-stream'
-    };
-    return headers;
+export async function UploadAndValidateHeader(formData: any) {
+  let headers: IHeaders = {
+    Authorization: "Bearer " + token,
+    "content-type": "application/octet-stream",
+  };
+  return headers;
 }
 export function dataPlaneHeader() {
-    let headers: IHeaders = {
-        'Authorization': 'Bearer '+ token
-    };
-    return headers;
-}
-export function startTestData(testRunName:string, runDisplayName: string, runDescription: string) {
-    var data = {
-        testRunId: testRunName,
-        displayName: runDisplayName ? runDisplayName : getDefaultTestRunName(),
-        description: runDescription ? runDescription : 'Started using GitHub Actions',
-        testId: testId,
-        secrets: secretsRun,
-        environmentVariables: envRun
-    };
-    return data;
+  let headers: IHeaders = {
+    Authorization: "Bearer " + token,
+  };
+  return headers;
+}
+export function startTestData(
+  testRunName: string,
+  runDisplayName: string,
+  runDescription: string
+) {
+  var data = {
+    testRunId: testRunName,
+    displayName: runDisplayName ? runDisplayName : getDefaultTestRunName(),
+    description: runDescription
+      ? runDescription
+      : "Started using GitHub Actions",
+    testId: testId,
+    secrets: secretsRun,
+    autoStopCriteria: autoStop,
```

Comment:
nit: Test run will take the value from test so should we remove it from here?

Reply:
Done